### PR TITLE
bump gems

### PR DESCRIPTION
**Pull request by grosser**
... and fix missing tests

@zendesk/compute

### CODE REVIEW

File: gem/samson_secret_puller.gemspec [hunk 1]
```diff
@@ -7,5 +9,5 @@ Gem::Specification.new name, "1.1.1" do |s|
   s.homepage = "https://github.com/zendesk/#{name}"
   s.files = `git ls-files lib MIT-LICENSE.txt`.split("\n")
   s.license = "MIT"
-  s.required_ruby_version = '>= 1.9.3'
+  s.required_ruby_version = '>= 2.4'
```
Comment (adammw) [inline gem/samson_secret_puller.gemspec]: probably needs a minor bump to signify the drop in ruby <2.3 compat.

File: lib/secrets.rb [hunk 2]
```diff
@@ -1,30 +1,24 @@
+# frozen_string_literal: true
+
 require 'vault'
 require 'openssl'
 require 'fileutils'
 
-# fixed in vault server 0.6.2 https://github.com/hashicorp/vault/pull/1795
```
Comment (adammw) [inline lib/secrets.rb]: do you keep a changelog for this gem? also good to note dropping support for <0.6.2

Comment (grosser) [reply lib/secrets.rb]: that's not in the gem, just the server

Comment (adammw) [reply lib/secrets.rb]: as in, you no longer patch around old versions of the server... no?

Comment (grosser) [reply lib/secrets.rb]: ah yes ... I though this was patching the gem not the server dug ... I'll bring that back, need more testing to remove that

**Pull request merged**